docs/howto/how_to_bulk_transform.rst

```diff
@@ -0,0 +1,118 @@
+
+Sources and Destinations
+------------------------
+
+For most sources and destinations we use `Apache Libcloud Object storage <https://libcloud.readthedocs.io/en/stable/supported_providers.html>`__ . This should cover casees from local storage to Amazon S3, MinIO, and many other cloud storage solutions. The ``--source_provider`` and ``--destination_provider`` options are the Libcloud Provider names, in all caps (ex: ``S3``, ``LOCAL``, ``MINIO``). The ``--source_config`` and ``--destination_config`` options are JSON strings passed directly to the Libcloud constructor as keyword args.
+
+The ``LRS`` destination provider is a special case that uses the usual event-routing-backends logic for sending events to Caliper and/or xAPI learning record stores.
+
+For the ``LOCAL`` provider, the path to the file(s) is a concatenation of the ``key``, which is the path to a top level directory, a ``container`` which is a single subdirectory name inside the ``key`` directory, and a ``prefix`` (if provided) will be appended to the container to determine the final path.
+
+::
+
+    # This will attempt to recursively read all files in the "/openedx/data/logs/" directory
+    {"key": "/openedx/data", "container": "logs"}
+
+    # This will attempt to recursively read all files in "/openedx/data/logs/tracking/" as
+    # well as any files in "/openedx/data/logs/" that begin with "tracking"
+    {"key": "/openedx/data", "container": "logs", "prefix": "tracking"}
+
+    # This will attempt to read a single file named "/openedx/data/logs/tracking.log"
+    {"key": "/openedx/data", "container": "logs", "prefix": "tracking.log"}
+
+
+For other providers ``key`` and ``secret`` are authentication credentials and ``container`` is roughly synonymous with an S3 bucket. Configuration for each provider is different, please consult the libcloud docs for your provider to learn about other options you may need to pass in to the ``--source_config`` and ``--destination_config`` JSON structures.
+
+
+Modes Of Operation
+------------------
+
+The command can work in a few distinct ways.
+
+**File(s) to learning record store (LRS)** - this will use the existing event-routing-backends configuration to route any log replays to **all** configured LRS backends just like the event was being emitted right now. This can be used to backfill old data, capture old events that didn't previously have transforms, or fix up lost data from downtime issues.
+
+.. note:: Note that any event filters in the ERB configuration will work as usual on these events.
+
+**File(s) to file(s)** - This will perform the same transformations as usual, but instead of routing them to an LRS they can be saved as a file to any libcloud destination. In this mode all events are saved to a single file and no filters are applied.
+
+.. note:: Note that NO event filters from the usual ERB configuration will impact these events, since they happen on a per-destination basis.
```

Comment:
Actually backend filters are also skipped since they happen at the tracking log emission level and we're working above there at this point. I'll try to clarify it all.

Reply:
Perfect, thank you for clarifying @bmtcril !